add docs for format: sql unit testing (#5281)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-properties/data-formats.md

        format: csv
        fixture: my_model_a_fixture

```

## sql

> This format:

Reviewer: ```suggestion
Using this format:
```

> - provides more flexbility for the types of data you can unit test

Reviewer: ```suggestion
- Provides more flexibility for the types of data you can unit test
```

> - allows you to unit test a model that depends on an ephemeral model

Reviewer: ```suggestion
- Allows you to unit test a model that depends on an ephemeral model
```

However, when using `format: sql` you must supply mock data for _all rows_.

When using the `sql` format, you can use either an inline SQL query for `rows`:

```yml

unit_tests:
  - name: test_my_model
    model: my_model
    given:
      - input: ref('my_model_a')
        format: csv
        rows: |
          select 1 as id, 'gerda' as name, null as loaded_at union all
          select 2 as id, 'michelle', null as loaded_at as name

```

Or, you can provide the name of a SQL file in the `tests/fixtures` directory (or the configured `test-paths` location) of your project for `fixture`: 

```yml

unit_tests:
  - name: test_my_model
    model: my_model
    given:
      - input: ref('my_model_a')
        format: sql
        fixture: my_model_a_fixture

```

> Note: Jinja is not supported in SQL fixtures for unit tests.

Reviewer: ```suggestion
**Note:** Jinja is unsupported in SQL fixtures for unit tests.
```
